### PR TITLE
DEPR: DTA(float_data, dtype=dt64tz)

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/dtypes/test_missing.py [hunk 1]
```diff
@@ -444,7 +442,7 @@ def test_array_equivalent(dtype_equal):
     )
     # The rest are not dtype_equal
```
Comment (mroeschke) [inline pandas/tests/dtypes/test_missing.py:443]: Nit: Could update this to say the second case is equal?

**Review by mroeschke — COMMENTED** [review]
One minor comment and merge conflict otherwise looks good

File: pandas/tests/dtypes/test_missing.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/dtypes/test_missing.py:443]: i think this is in reference to the dtype_equal keyword

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Comment by phofl** [comment]
@jbrockmendel looks like the arm build started failing after this one. Is this possible?

**Comment by jbrockmendel** [comment]
its plausible.  IIRC there was something about `np.array([np.nan]).astype(np.int64)` behaving weird on ARM.

**Comment by jbrockmendel** [comment]
could try changing `data = data.astype(np.int64)` to `data = data.astype("datetime64[ns]").view("i8")` and see if that does the trick

**Pull request merged**